server/market/market.go

```diff
@@ -3,30 +3,251 @@
 
 package market
 
-// The Market[Manager] will:
-// - Cycle the epochs. Epoch timing, the current epoch index, etc.
-// - Manage multiple epochs (one active and others in various states of
-//   matching, swapping, or archival).
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/decred/dcrdex/server/asset"
+	"github.com/decred/dcrdex/server/book"
+	"github.com/decred/dcrdex/server/db"
+	"github.com/decred/dcrdex/server/market/types"
+	"github.com/decred/dcrdex/server/matcher"
+	"github.com/decred/dcrdex/server/order"
+	"github.com/decred/dcrdex/server/swap"
+)
+
+// Error is just a basic error.
+type Error string
+
+// Error satisfies the error interface.
+func (e Error) Error() string {
+	return string(e)
+}
+
+// ErrClientDisconnected will be returned if Send or Request is called on a
+// disconnected link.
+const ErrMarketNotRunning = Error("market not running")
+const ErrInvalidOrder = Error("order failed validation")
+
+type bookUpdateAction uint8
+
+const (
+	invalidAction bookUpdateAction = iota
+	epochAction
+	bookAction
+	unbookAction
+)
+
+type BookUpdateSignal struct {
+	action bookUpdateAction
+	order  order.Order
+}
```

Comment:
`BookUpdateSignal` and `bookUpdateAction` correspond to the types in PR #62 @ https://github.com/decred/dcrdex/pull/62/files/108e21719b6416eb5d89f694e2cefe08c5db1dc8#diff-2440117baeda2a1269b5c1c8b352437dR19-R41

Reply:
Did you need this exported, @chappjc?

Reply:
If it's in the `market` package, I guess not.  I can always change it.